plugin/PySrc/mock_turtle.py

```diff
@@ -1023,3 +1026,52 @@ def _color(self, colorstr):
     'yellow4': '#8b8b00',
     'yellowgreen': '#9acd32',
 }
+
+
+class MockPyglet(object):
+
+    class Window(pyglet.window.Window):
+
+        def __init__(self, **kwargs):
+            conf = pyglet.gl.Config(double_buffer=True)
+            super(MockPyglet.Window, self).__init__(
+                config=conf,
+                resizable=True,
+                visible=False,
+                width=self.width,
+                height=self.height
+            )
+            self.on_resize(self.width, self.height)
+
+        def on_draw(self):
+
+            # Get the colour buffer, write it to a bytearray in png format.
+            buf = pyglet.image.get_buffer_manager().get_color_buffer()
+            b = io.BytesIO()
+            buf.save('buffer.png', b)
```

Comment:
Have you checked to see if the contents of this buffer actually contain a useful image? After I fixed the base64 encoding problems that were breaking PyCharm, I don't see any image.